pr.js

```diff
@@ -0,0 +1,307 @@
+          } else {
+            todoLi.textContent = '( ) ' + element.todoText;
+          };
+          todoLi.appendChild(view.createDeleteButton());
+          todoLi.appendChild(view.createToggleButton());
+          todoLi.appendChild(view.changeTodoInputField()); 
+          newTodosUl.appendChild(todoLi);
+        })
+        var writeHere = document.getElementById(i);
+        writeHere.appendChild(newTodosUl)
+      }
+    }
+  },
+
+  createDeleteButton: function() {
+    var deleteButton = document.createElement('button');
+    deleteButton.textContent = 'Delete';
+    deleteButton.className = 'deleteButton';
+    return deleteButton;
+  },
+
+  createToggleButton: function() {
+    var toggleButton = document.createElement('button');
+    toggleButton.textContent = 'Toggle';
+    toggleButton.className = 'toggleButton';
+    return toggleButton;
+  },
+
+  changeTodoInputField: function() {
+    var todoInputField = document.createElement('input');
+    todoInputField.type = 'text';
+    todoInputField.className = 'todoInput';
+    return todoInputField; 
+  },
+
+  eventListeners: function() {
+    var todosUl = document.querySelector('ul');
+    todosUl.addEventListener('click', function(event) {
+      if (event.target.className === 'deleteButton') {
+        var position = parseInt(event.target.parentNode.id)
+        var positionMap = [];
+        var eventPlaceholder = event.target
+        function positionMapper(eventPlaceholder){
+          if (eventPlaceholder.parentNode.id === 'headOfTheHouse') {           
+            return positionMap;
+          } else {
+            var position = parseInt(eventPlaceholder.parentNode.id);
+            positionMap.unshift(position);
+            //move up one level
+            eventPlaceholder = eventPlaceholder.parentNode;
+            return positionMapper(eventPlaceholder);
+          }
+        }
+        positionMapper(eventPlaceholder);
+        handlers.deleteTodo(positionMap);
+      }
+      if (event.target.className === 'toggleButton') {
+        var position = parseInt(event.target.parentNode.id)
+        handlers.toggleCompleted(position);
+      }
+    });
+    todosUl.addEventListener('keyup', function(event) {
+      if (event.target.className === 'todoInput' && event.keyCode === 13) {
+        var position = parseInt(event.target.parentNode.id);
+        var inputFieldValue = event.srcElement.value;
+        handlers.changeTodo(position, inputFieldValue); 
+      }
+    });
+    todosUl.addEventListener('keyup', function(event) {
+      if (event.target.className === 'todoInput' && event.keyCode === 40) {
+        // var inputFieldValue = event.srcElement.value;
+        // var position = parseInt(event. target.parentNode.id);
+        // handlers.nestTodo(position, inputFieldValue);
+        var inputFieldValue = event.srcElement.value;
+        var positionMap = [];
+        var eventPlaceholder = event.target
+        function positionMapper(eventPlaceholder){
```

Comment:
whenever you have a particularly complex piece of code, it's generally nice to explain the general idea of it in prose. it gives readers the chance to understand what you're trying to achieve from a high-level before trying to parse it line-by-line.

things like ASCII art are super helpful too. or you could show a representation of a positionmap for a given todos list. etc. ex:
```
position map [1,2,1] would be the second. top level todo, then the next nested todo would be the third, and the next nested todo would be the second.
```

hi-level comments tend to age much better than specific ones.